### ISSUE TITLE
Upgrade to .NET Core SDK 2.0.2

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#1247]
This should fix #1197, and have better support for VB.NET projects.

**Comment by thomaslevesque** [comment]
In theory, this should be as easy as changing `1.0.3` to `2.0.0` in `global.json`, but for some reason, it causes some tests to fail... All tests in `TypeCatalogueTests` fail with this error when targeting net461:

```

System.AggregateException : One or more errors occurred.
---- System.IO.FileNotFoundException : Could not load file or assembly 'System.Security.Cryptography.Algorithms, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
-------- System.IO.FileNotFoundException : Could not load file or assembly 'System.Security.Cryptography.Algorithms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
---- The following constructor parameters did not have matching fixture data: ExternalAssemblyGenerator externalAssemblyGenerator

----- Inner Stack Trace #1 (System.IO.FileNotFoundException) -----
   at Microsoft.CodeAnalysis.CryptographicHashProvider.ComputeSha1(Byte[] bytes)
   at Microsoft.CodeAnalysis.AssemblyIdentity.CalculatePublicKeyToken(ImmutableArray`1 publicKey)
   at Microsoft.CodeAnalysis.AssemblyIdentity.get_PublicKeyToken()
   at Microsoft.CodeAnalysis.AssemblyIdentity.KeysEqual(AssemblyIdentity x, AssemblyIdentity y)
   at Microsoft.CodeAnalysis.AssemblyIdentity.MemberwiseEqual(AssemblyIdentity x, AssemblyIdentity y)
   at Microsoft.CodeAnalysis.AssemblyIdentityComparer.TriviallyEquivalent(AssemblyIdentity x, AssemblyIdentity y)
   at Microsoft.CodeAnalysis.AssemblyIdentityComparer.Compare(AssemblyIdentity reference, String referenceDisplayName, AssemblyIdentity definition, Boolean& unificationApplied, Boolean ignoreVersion)
   at Microsoft.CodeAnalysis.CommonReferenceManager`2.ResolveReferencedAssembly(AssemblyIdentity reference, ImmutableArray`1 definitions, Int32 definitionStartIndex, AssemblyIdentityComparer assemblyIdentityComparer)
   at Microsoft.CodeAnalysis.CommonReferenceManager`2.ResolveReferencedAssemblies(ImmutableArray`1 references, ImmutableArray`1 definitions, Int32 definitionStartIndex, AssemblyIdentityComparer assemblyIdentityComparer)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.ReferenceManager.AssemblyDataForMetadataOrCompilation.BindAssemblyReferences(ImmutableArray`1 assemblies, AssemblyIdentityComparer assemblyIdentityComparer)
   at Microsoft.CodeAnalysis.CommonReferenceManager`2.Bind(TCompilation compilation, ImmutableArray`1 explicitAssemblies, ImmutableArray`1 explicitModules, ImmutableArray`1 explicitReferences, ImmutableArray`1 explicitReferenceMap, MetadataReferenceResolver resolverOpt, MetadataImportOptions importOptions, Boolean supersedeLowerVersions, Dictionary`2 assemblyReferencesBySimpleName, ImmutableArray`1& allAssemblies, ImmutableArray`1& implicitlyResolvedReferences, ImmutableArray`1& implicitlyResolvedReferenceMap, DiagnosticBag resolutionDiagnostics, Boolean& hasCircularReference, Int32& corLibraryIndex)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.ReferenceManager.CreateAndSetSourceAssemblyFullBind(CSharpCompilation compilation)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.ReferenceManager.CreateSourceAssemblyForCompilation(CSharpCompilation compilation)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetRuntimeMetadataVersion(EmitOptions emitOptions)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetRuntimeMetadataVersion(EmitOptions emitOptions, DiagnosticBag diagnostics)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CreateModuleBuilder(EmitOptions emitOptions, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, IEnumerable`1 manifestResources, CompilationTestData testData, DiagnosticBag diagnostics, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Compilation.CheckOptionsAndCreateModuleBuilder(DiagnosticBag diagnostics, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.FileSystemExtensions.Emit(Compilation compilation, String outputPath, String pdbPath, String xmlDocPath, String win32ResourcesPath, IEnumerable`1 manifestResources, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpFileSystemExtensions.Emit(CSharpCompilation compilation, String outputPath, String pdbPath, String xmlDocumentationPath, String win32ResourcesPath, IEnumerable`1 manifestResources, CancellationToken cancellationToken)
   at FakeItEasy.IntegrationTests.ExternalAssemblyGenerator.EmitDependencyAssembly() in D:\Git\FakeItEasy\tests\FakeItEasy.IntegrationTests\ExternalAssemblyGenerator.cs:line 96
   at FakeItEasy.IntegrationTests.ExternalAssemblyGenerator..ctor() in D:\Git\FakeItEasy\tests\FakeItEasy.IntegrationTests\ExternalAssemblyGenerator.cs:line 20
----- Inner Stack Trace -----

----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
```

I don't understand it... we're only upgrading the SDK, not the frameworks we're targeting, so there should be no difference at all.

**Comment by thomaslevesque** [comment]
Adding a package reference to `System.Security.Cryptography.Algorithms` has no effect; the assembly isn't copied to the output folder. Even explicitly using a type from that assembly (`IncrementalHash`) to ensure that the assembly reference is kept has no effect either. The only thing that works is manually copying the assembly to the output folder 😕.

**Comment by thomaslevesque** [comment]
Seems related to https://github.com/dotnet/sdk/issues/1410, which is supposed to be fixed...
Adding `<DependsOnNETStandard>True</DependsOnNETStandard>` to FakeItEasy.IntegrationTests.csproj causes the assembly to be copied to the output folder, but it also copies all other .NET Standard assemblies...

**Comment by thomaslevesque** [comment]
OK, I hadn't realized there was a newer version of the SDK... it works fine with 2.0.2.

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.2.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/4.2.0).
